tests/tests_geomstats/test_hypersphere.py

```diff
@@ -1,911 +1,701 @@
+                is_tangent_atol=gs.atol * 1000,
+                atol=gs.atol * 1000,
+            )
 
-    def test_angle_to_extrinsic(self):
-        space = Hypersphere(1)
-        point = gs.pi / 4
-        result = space.angle_to_extrinsic(point)
-        expected = gs.array([1.0, 1.0]) / gs.sqrt(2.0)
-        self.assertAllClose(result, expected)
+        def exp_and_dist_and_projection_to_tangent_space_data(self):
+            unnorm_base_point = gs.array([16.0, -2.0, -2.5, 84.0, 3.0])
+            base_point = unnorm_base_point / gs.linalg.norm(unnorm_base_point)
+            smoke_data = [
+                dict(
+                    dim=4,
+                    vector=gs.array([9.0, 0.0, -1.0, -2.0, 1.0]),
+                    base_point=base_point,
+                )
+            ]
+            return self.generate_tests(smoke_data)
 
-        point = gs.array([1.0 / 3, 0.0]) * gs.pi
-        result = space.angle_to_extrinsic(point)
-        expected = gs.array([[1.0 / 2, gs.sqrt(3.0) / 2], [1.0, 0.0]])
-        self.assertAllClose(result, expected)
+    testing_data = TestDataHypersphereMetric()
 
-    def test_extrinsic_to_angle(self):
-        space = Hypersphere(1)
-        point = gs.array([1.0, 1.0]) / gs.sqrt(2.0)
-        result = space.extrinsic_to_angle(point)
-        expected = gs.pi / 4
+    def test_inner_product(
+        self, dim, tangent_vec_a, tangent_vec_b, base_point, expected
+    ):
+        metric = self.metric(dim)
+        result = metric.inner_product(
+            gs.array(tangent_vec_a), gs.array(tangent_vec_b), gs.array(base_point)
+        )
         self.assertAllClose(result, expected)
 
-        point = gs.array([[1.0 / 2, gs.sqrt(3.0) / 2], [1.0, 0.0]])
-        result = space.extrinsic_to_angle(point)
-        expected = gs.array([1.0 / 3, 0.0]) * gs.pi
+    def test_dist(self, dim, point_a, point_b, expected):
+        metric = self.metric(dim)
+        result = metric.dist(gs.array(point_a), gs.array(point_b))
+        self.assertAllClose(result, gs.array(expected))
+
+    def test_dist_pairwise(self, dim, point, expected, rtol):
+        metric = self.metric(dim)
+        result = metric.dist_pairwise(gs.array(point))
+        self.assertAllClose(result, gs.array(expected), rtol=rtol)
+
+    def test_diameter(self, dim, points, expected):
+        metric = self.metric(dim)
+        result = metric.diameter(gs.array(points))
+        self.assertAllClose(result, gs.array(expected))
+
+    def test_christoffels_shape(self, dim, point, expected):
+        metric = self.metric(dim)
+        result = metric.christoffels(point)
+        self.assertAllClose(gs.shape(result), expected)
+
+    def test_sectional_curvature(
+        self, dim, tangnet_vec_a, tangent_vec_b, base_point, expected
+    ):
+        metric = self.metric(dim)
+        result = metric.sectional_curvature(tangnet_vec_a, tangent_vec_b, base_point)
+        self.assertAllClose(result, expected, atol=1e-5)
```

Comment:
atol should be a parameter of the test function?

Reply:
could be... if really needs to depend on actual individual data point.